internal/controller/profiles/common/object_creators.go

```diff
@@ -439,10 +443,37 @@ func UserPropsConfigMapCreator(workflow *operatorapi.SonataFlow) (client.Object,
 
 // ManagedPropsConfigMapCreator creates an empty ConfigMap to hold the external application properties
 func ManagedPropsConfigMapCreator(workflow *operatorapi.SonataFlow, platform *operatorapi.SonataFlowPlatform) (client.Object, error) {
-
 	props, err := properties.ApplicationManagedProperties(workflow, platform)
 	if err != nil {
 		return nil, err
 	}
 	return workflowproj.CreateNewManagedPropsConfigMap(workflow, props), nil
 }
+
+// ServiceMonitorCreator is an ObjectsCreator for Service Monitor for the workflow service.
+func ServiceMonitorCreator(workflow *operatorapi.SonataFlow) (client.Object, error) {
+	lbl := workflowproj.GetMergedLabels(workflow)
+	spec := &prometheus.ServiceMonitorSpec{
+		Selector: metav1.LabelSelector{
+			MatchLabels: map[string]string{
```

Comment:
@wmedvede The selector defined in the servicemonitor's spec is used for Prometheus to identify the k8s services that it uses to scrape the metrics, and hence I'd like to keep the selector with the minimal set of the labels, and adding the remaining labels as part of the selector might have some performance impact on the Prometheus scraping process. My preference is to keep it as it is. What do you think?

Reply:
IDK why it should introduce performance issues, when all the k8s platform is designed to fast queries based on labels, but for me it's fine.

Reply:
@wmedvede While label queries on k8s is "fast", the additional labels added would require more computation. Since  Prometheus scrapes metrics periodically, this would cause unnecessary accumulative cost in production from time to time.